Fix:

error: reference to unresolved using declaration
error: excess elements in scalar initializer

by removing base_aligned_alloc because it calls aligned_alloc which macOS 10.14
and earlier don't have. Fortunately nothing calls base_aligned_alloc yet.
--- aseprite/laf/base/memory.cpp.orig	2023-11-15 13:30:37.000000000 -0600
+++ aseprite/laf/base/memory.cpp	2024-01-11 20:08:06.000000000 -0600
@@ -299,14 +299,6 @@
 
 #endif
 
-void* base_aligned_alloc(std::size_t bytes, std::size_t alignment)
-{
-#if LAF_WINDOWS
-  return _aligned_malloc(bytes, alignment);
-#else
-  return aligned_alloc(alignment, bytes);
-#endif
-}
 
 void base_aligned_free(void* mem)
 {
--- aseprite/laf/base/memory.h.orig	2023-11-15 13:30:37.000000000 -0600
+++ aseprite/laf/base/memory.h	2024-01-11 20:10:39.000000000 -0600
@@ -34,11 +34,6 @@
 void  base_free   (void* mem);
 char* base_strdup (const char* string);
 
-#ifdef __cplusplus
-  void* base_aligned_alloc(std::size_t bytes, std::size_t alignment = base_alignment);
-#else
-  void* base_aligned_alloc(std::size_t bytes, std::size_t alignment);
-#endif
 void base_aligned_free(void* mem);
 
 #ifdef LAF_MEMLEAK
